lib/mempool/mempool.js

```diff
@@ -1659,13 +1741,39 @@ class Mempool extends EventEmitter {
   isDoubleSpend(tx) {
     for (const {prevout} of tx.inputs) {
       const {hash, index} = prevout;
-      if (this.isSpent(hash, index))
+      const conflict = this.getSpent(hash, index);
+
+      if (conflict && !conflict.tx.isRBF())
         return true;
     }
 
     return false;
   }
 
+  /**
+   * Get an array of all transactions currently in the mempool that
+   * spend one or more of the same outputs as an incoming transaction.
+   * @param {TX} tx
+   * @returns {Promise} - Returns (@link MempoolEntry}[].
+   */
+
+  getConflicts(tx) {
+    const conflicts = [];
+    const hashes = new BufferSet();
+
+    for (const {prevout} of tx.inputs) {
+      const {hash, index} = prevout;
+      const conflict = this.getSpent(hash, index);
+
+      if (conflict && !hashes.has(hash)) {
+        hashes.add(hash);
+        conflicts.push(conflict);
+      }
+    }
+
+    return conflicts;
+  }
```

Comment:
why not just do this?
```suggestion
function getConflicts(tx) {
  const conflicts = new Set();
  
  for (const { prevout: { hash, index } } of tx.inputs) {
    const conflict = this.getSpent(hash, index);
    
    if (conflict) {
      conflicts.add(conflict);
    }
  }

  return Array.from(conflicts);
}
```

Reply:
great patch, thanks